enhancements/installer/single-node-installation-bootstrap-in-place.md

```diff
@@ -0,0 +1,335 @@
+3. Cannot use Bare Metal IPI:
+   a. Adds irrelevant dependencies - vips, keepalived, mdns
+   b. Requires same L2 between bootstrap and the Single Node OpenShift
+
+### Goals
+
+* Describe an approach for installing Single Node OpenShift in a BareMetal environment for production use.
+* Minimal changes to OpenShift installer and the implementation shouldn't affect existing deployment flows.
+* Installation should result a clean Single Node OpenShift without any bootstrap leftovers.
+
+
+### Non-Goals
+
+* Addressing a similar installation flow for multi-node clusters.
+* High-availability for Single Node OpenShift.
+* Single-node-developer (CRC) cluster-profile installation.
+* Supporting cloud deployment for bootstrap in place (since livecd cannot be used). It will be addressed as part of future enhancement.
+* Upgrading Single Node OpenShift will be addressed as part of a future enhancement.
+
+## Proposal
+
+The installer will be enhanced to provide a way to generate a single node ignition configuration. 
+The user will be able to boot a RHCOS live CD with that ignition to initiate the installation.
+The live CD will perform the cluster bootstrap flow.
+A master ignition including the control plane static pods will be created as part of the bootstrap. The master ignition will then be used on reboot to complete the installation and bring up Single Node OpenShift.
+Use of the liveCD helps to ensure that we have a clean bootstrap
+flow with just the master ignition as the handoff point.
+
+
+### User Stories
+
+#### As an OpenShift user, I want to be able to deploy OpenShift on a supported single node configuration
+
+A user will be able to run the OpenShift installer to create a single-node
+deployment, with some limitations (see non-goals above). The user
+will not require special support exceptions to receive technical assistance
+for the features supported by the configuration.
+
+### Implementation Details/Notes/Constraints
+
+The OpenShift installer `create ignition-configs` command will generate a `bootstrap-in-place-for-live-iso.ign`
+file, when the number of replicas for the control plane (in the `install-config.yaml`) is `1`.
+
+The `bootstrap-in-place-for-live-iso.ign` will be embedded into an RHCOS liveCD by the user using the `coreos-install embed` command.
+
+The user will boot a machine with this liveCD and the liveCD will start executing a similar flow to a bootstrap node in a regular installation.
+
+`bootkube.sh` running on the live ISO will execute the rendering logic.
+The live ISO environment provides a scratch place to write bootstrapping files so that they don't end up on the real node. This eliminates a potential source of errors and confusion when debugging problems.
+
+The bootstrap static pods will be generated in a way that the control plane operators will be able to identify them and either continue in a controlled way for the next revision, or just keep them as the correct revision and reuse them.
```

Comment:
This statement sounds like we know what we'll do, or we're assuming we can do it but haven't actually worked out the details. Maybe that's my lack of insight into how those static pods are defined? Is this just a matter of the operators looking at some files with pod definitions? Or are they going to have to do something more complicated?

Reply:
For now it's the 2nd option - operators are looking into the static manifests and updating the revisions. From that point, we can only improve: make the control plane operators be aware of this transition and dont update a revision? spawn parallel kubeapi and update gradually - all should be improvements farther the road

Reply:
@hexfusion I think what you mentioned [here](https://github.com/openshift/enhancements/pull/565#discussion_r553021594)
Is discussed here.